### PR TITLE
adding implementation details to geometryConverters

### PR DESCRIPTION
**Pull request by TianJingwd**
## What is the change?

This PR adds additional implementation details on the impl tags within geometryConverters

## Why is the change being made?

SQA work

---

## Checklist

- [x] This PR has only [one purpose or idea](https://terrapower.github.io/armi/developer/tooling.html#one-idea-one-pr).
- [x] [Tests](https://terrapower.github.io/armi/developer/tooling.html#test-it) have been added/updated to verify that the new/changed code works.

- [x] The code style follows [good practices](https://terrapower.github.io/armi/developer/standards_and_practices.html).
- [x] The commit message(s) follow [good practices](https://terrapower.github.io/armi/developer/tooling.html).

- [x] The [release notes](https://terrapower.github.io/armi/release/index.html) (location `doc/release/0.X.rst`) are up-to-date with any important changes.
- [x] The [documentation](https://terrapower.github.io/armi/developer/tooling.html#document-it) is still up-to-date in the `doc` folder.
- [x] If any [requirements](https://terrapower.github.io/armi/developer/tooling.html#watch-for-requirements) were affected, mention it in the [release notes](https://terrapower.github.io/armi/release/index.html).
- [x] The dependencies are still up-to-date in `pyproject.toml`.

### CODE REVIEW

File: armi/reactor/converters/geometryConverters.py [hunk 1]
```diff
@@ -1255,6 +1262,20 @@ def convert(self, r):
             :id: I_ARMI_THIRD_TO_FULL_CORE0
             :implements: R_ARMI_THIRD_TO_FULL_CORE
 
+
```
Comment (albeanth) [inline armi/reactor/converters/geometryConverters.py:1265]:
```suggestion
```

File: armi/reactor/converters/geometryConverters.py [hunk 2]
```diff
@@ -1255,6 +1262,20 @@ def convert(self, r):
...
+            This method first checks if the input model is already full core.
```
Comment (albeanth) [inline armi/reactor/converters/geometryConverters.py:1266]:
```suggestion
            This method first checks if the input reactor is already full core.
```

File: armi/reactor/converters/geometryConverters.py [hunk 3]
```diff
@@ -1255,6 +1262,20 @@ def convert(self, r):
...
+            If full-core symmety is detected, the source model will be returned.
```
Comment (albeanth) [inline armi/reactor/converters/geometryConverters.py:1267]:
```suggestion
            If full-core symmetry is detected, the input reactor is returned.
```

File: armi/reactor/converters/geometryConverters.py [hunk 4]
```diff
@@ -1255,6 +1262,20 @@ def convert(self, r):
...
+            If not, it then verifies that the input model has the expected one-third
```
Comment (albeanth) [inline armi/reactor/converters/geometryConverters.py:1268]:
```suggestion
            If not, it then verifies that the input reactor has the expected one-third
```

**Review by albeanth — COMMENTED** [review]
Just a couple of comments before handing this over to @opotowsky .

File: armi/reactor/converters/geometryConverters.py [hunk 5]
```diff
@@ -434,6 +434,13 @@ def convert(self, r):
             :id: I_ARMI_CONV_3DHEX_TO_2DRZ
             :implements: R_ARMI_CONV_3DHEX_TO_2DRZ
 
+            This method converts the hex-z mesh to r-theta-z mesh.
+            It first verifies that the geometry type of the input reacotr ``r``
+            has the expected HEX geomety. Upon conversion, it determines the inner
+            and outer diameters of each rings in the r-theta-z mesh and calls
+            ``_createRadialThetaZone`` to create radial theta zone with homogenized mixture.
```
Comment (opotowsky) [inline armi/reactor/converters/geometryConverters.py:441]:
```suggestion
            It first verifies that the geometry type of the input reactor ``r``
            has the expected HEX geometry. Upon conversion, it determines the inner
            and outer diameters of each ring in the r-theta-z mesh and calls
            ``_createRadialThetaZone`` to create a radial theta zone with a homogenized mixture.
```

File: armi/reactor/converters/geometryConverters.py [hunk 6]
```diff
@@ -1346,6 +1367,11 @@ def restorePreviousGeometry(self, r=None):
         .. impl:: Restore a one-third-core geometry to a full-core geometry.
             :id: I_ARMI_THIRD_TO_FULL_CORE1
             :implements: R_ARMI_THIRD_TO_FULL_CORE
+
+            This method is a reverse process of the method ``convert``. It converts
+            the full-core model back to the original one-third core model by removing
```
Comment (opotowsky) [inline armi/reactor/converters/geometryConverters.py:1372]:
```suggestion
            the full-core reactor model back to the original one-third core reactor model by removing
```
Along the lines of tony's edits above.

File: armi/reactor/converters/geometryConverters.py [hunk 7]
```diff
@@ -1458,6 +1489,10 @@ def removeEdgeAssemblies(self, core):
             :id: I_ARMI_ADD_EDGE_ASSEMS1
             :implements: R_ARMI_ADD_EDGE_ASSEMS
 
+            This method is the reverse process of the method ``addEdgeAssemblies``. It is
+            needed for the DIF3D-Nodal calculation. It removes the assemblies on the 120-degree
+            symmety line.
```
Comment (opotowsky) [inline armi/reactor/converters/geometryConverters.py:1494]:
```suggestion
            symmetry line.
```

File: armi/reactor/converters/geometryConverters.py [hunk 8]
```diff
@@ -1388,6 +1414,11 @@ def addEdgeAssemblies(self, core):
             :id: I_ARMI_ADD_EDGE_ASSEMS0
             :implements: R_ARMI_ADD_EDGE_ASSEMS
 
+            Edge assemblies on the 120-degree symmetric line of one-third core model are added
```
Comment (opotowsky) [inline armi/reactor/converters/geometryConverters.py:1417]:
```suggestion
            Edge assemblies on the 120-degree symmetric line of a one-third core reactor model are added
```

**Review by opotowsky — COMMENTED** [review]
I think things read really well and represent the code. Just a few typos!

**Review by opotowsky — APPROVED** [review]
All done -- great work and thank you @TianJingwd !

**Pull request merged**